content/tutorials/build-apps/build-a-desktop-wallet-in-js.md

```diff
@@ -0,0 +1,231 @@
+
+You can find the complete source code for all of this tutorial's examples in the [code samples section of this website's repository]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/).
+
+## Goals
+
+At the end of this tutorial, you should be able to build a simple XRP wallet displayed below.
+
+![Home Page Screenshot](img/js-wallet-home.png)
+
+This application can:
+
+- Show updates to the XRP Ledger in real-time.
+- View any XRP Ledger account's activity, including showing how much XRP was delivered by each transaction.
+- Show how much XRP is set aside for the account's [reserve requirement](reserves.html). 
+- Send [direct XRP payments](direct-xrp-payments.html), and provide feedback about the intended destination address, including:
+    - Displaying your account's available balance
+    - Verifying that the destination address is valid
+    - Validating the account has enough XRP to send
+    - Allowing you to specify a destination tag
+
+## Steps
+
+Before you begin, make sure you have the pre-requisites installed. Check your node version by running `node -v`. If you don't have node installed, you can download it [here](https://nodejs.org/en/download/).
+
+### Step 1: Setting up the project
+
+1. Navigate to the directory that you want to create the project in.
+2. Run 
+   ```sh
+    yarn create vite-app <project-name>
+    ```
+    or
+    ```sh
+    npm init vite-app <project-name>
+    ```
+    to create a new project.
+
+3. Navigate to your project directory and copy this [package.json](_code-samples/build-a-wallet/js/package.json) file to your project.
+        - Alternatively you can also do `yarn add <package-name>` or `npm install <package-name>` for each individual package to add them to your `package.json` file.
+  
+4. Run `yarn` or `npm i` to install the packages.
+
+5. Create a new file `.env` in the root directory of the project and add the following variables:
+
+```bash
+CLIENT="wss://s.altnet.rippletest.net/" // or any other server
+EXPLORER_NETWORK="testnet" // or "mainnet"
+SEED="sEd7B2Jm7RBhhh5i88SH38aZB8ot8Kq" // Replace with your seed
+```
+6. Change the seed to your own seed. You can get a testnet account from [here](https://xrpl.org/xrp-test-net-faucet.html).
+
+### Step 2: Creating the Home Page (Displaying Account & Ledger Details)
+
+In this step, we will create the home page that will display the account details and the ledger details.
+
+![Home Page Screenshot](img/js-wallet-home.png)
+
+1. Make a new folder named src in the root directory of the project.
+
+2. Create new files named index.html, index.js and index.css. 
+
+3. Copy the contents of the [index.html](_code-samples/build-a-wallet/js/index.html), [index.js](_code-samples/build-a-wallet/js/index.js) and [index.css](_code-samples/build-a-wallet/js/index.css) files to your files.
+
+4. Create a new file in helpers folder named get-wallet-details.js and copy the code written below to the file.
```

Comment:
This is a lot of code to copy and paste - since this snippet is one that you're hoping the reader will be able to understand, I'd recommend taking a bit more time before showing the code to explain what we're about to do and why. 

Ex. 

"Now that you've got the basic setup, let's discuss what we're going to do. For the home page, we want to display our account info, what's happening on the ledger, and for fun add a little logo. In order to make that happen, first we need to look up our account state. This next bit of code handles that.

4. Create a new file in your `helpers` folder named `get-wallet-details.js` and copy the code written below into the file. "

...

(Later on before step 5)

"Now that we've got our account information, we want to look up the latest ledger to verify that the data we're showing people is up to date."

5. Create another file...

Reply:
You can totally explain the exact technical details of how it works after, but by giving people context on what's about to happen + why it matters they'll have a better understanding of the important pieces of the overall design :)